template/template.go

```diff
@@ -0,0 +1,197 @@
+package template
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"sync"
+
+	ast "go.expect.digital/mf2/parse"
+)
+
+type execFn func(operand any, opts map[string]any) (string, error)
+
+var execFuncs = map[string]execFn{}
+
+var mutex sync.Mutex
+
+// AddFunc adds a function to the template's function map.
+func AddFunc(name string, f func(any, map[string]any) (string, error)) {
```

Comment:
*Avoid package-level variables in this repo - singletons.*

If you have a project importing two packages using the same function name but different implementations, the templating is broken.

Add functions to `type Template` instead.